python-sdk/tutorials/automl-with-azureml/ONNX_batch_model_generator_automl_for_images.py

```diff
@@ -0,0 +1,200 @@
+            dynamic_axes['input'] = {0: 'batch'}
+            for output in output_names:
+                dynamic_axes[output] = {0: 'prediction'}
+
+            dummy_input = torch.randn(batch_size, 3, height_onnx, width_onnx).to(device)
+            model_wrapper.disable_model_transform()
+            model = model_wrapper.model
+
+    if batch_model:
+        try:
+            # export the model
+            export_onnx_model(model, dummy_input, input_names, output_names, dynamic_axes,
+                              onnx_model_file_path, device=device)
+            # check/verify schema of generated onnx model
+            onnx_model = onnx.load(onnx_model_file_path)
+            onnx.checker.check_model(onnx_model)
+            logger.info('ONNX model generation for the batch size {} is successful'.format(batch_size))
+        except Exception as e:
+            logger.error('ONNX model generation or Schema validation error: ' + str(e))
+            raise
+    else:
+        msg = 'Please use the auto-generated ONNX model for the best child run. No need to run this script'
+        logger.warning(msg)
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser()
+    # add arguments
+    parser.add_argument('--batch_size', type=int, required=True)
+    parser.add_argument('--height_onnx', type=int, required=True)
+    parser.add_argument('--width_onnx', type=int, required=True)
+    parser.add_argument('--experiment_name', type=str, required=True)
+    parser.add_argument('--subscription_id', type=str, required=True)
+    parser.add_argument('--resource_group', type=str, required=True)
+    parser.add_argument('--workspace_name', type=str, required=True)
+    parser.add_argument('--run_id', type=str, required=True)
+    parser.add_argument('--task_type', type=str, required=True)
+    parser.add_argument('--box_score_thresh', type=float, required=True)
+
+    parser.add_argument('--min_size', type=int, required=False)
+    parser.add_argument('--max_size', type=int, required=False)
+    parser.add_argument('--box_nms_thresh', type=float, required=False)
+    parser.add_argument('--box_detections_per_img', type=int, required=False)
+    parser.add_argument('--img_size', type=int, required=False)
+    parser.add_argument('--model_size', type=str, required=False)
+    parser.add_argument('--box_iou_thresh', type=float, required=False)
+
+    # parse arguments
+    args = parser.parse_args()
+
+    ws = Workspace.create(name=args.workspace_name,
+                          subscription_id=args.subscription_id,
+                          resource_group=args.resource_group,
+                          exist_ok=True)
+    experiment = Experiment(ws, name=args.experiment_name)
+
+    # load the best child
+    automl_image_run = AutoMLRun(experiment=experiment, run_id=args.run_id)
+    best_child_run = automl_image_run.get_best_child()
+
+    model_type = None
+    if args.task_type == 'image-object-detection':
```

Comment:
can we just have the task type sent in the script run be 'image-object-detection' and 'image-object-detection-yolo' instead of relying on img_size?

Reply:
sure